__tests__/comment.test.js

```diff
@@ -0,0 +1,103 @@
+import chai from 'chai';
+import chaiHttp from 'chai-http';
+import sinon from 'sinon';
+import models from '../database/models';
+
+import app from '../index';
+
+chai.use(chaiHttp);
+
+const url = '/api/v1/articles';
+const { expect } = chai;
+let validToken;
+
+describe('Testing comment endpoints', () => {
+  before((done) => {
+    const user = {
+      email: 'dd@test.com',
+      password: 'PasswoRD123__',
+    };
+    chai
+      .request(app)
+      .post('/api/v1/auth/signin')
+      .send(user)
+      .end((err, res) => {
+        const { token } = res.body.user;
+        validToken = token;
+        done();
+      });
+  });
+
+  it('should not post a comment on an article that doesnt exist', (done) => {
+    chai
+      .request(app)
+      .post(`${url}/slug-of-some-article-that-doesnt-exit-in-db/comments`)
+      .set('Authorization', `Bearer ${validToken}`)
+      .send({ body: 'sample comment body' })
+      .end((err, res) => {
+        expect(res.status).to.eql(404);
+        expect(res.body).to.have.property('errors');
+        expect(res.body.errors).to.have.property('message');
+        expect(res.body.errors.message).to.eql('Resource not found');
+        done();
+      });
+  });
+  it('should not post a comment on an article that is not yet published', (done) => {
+    chai
```

Comment:
If an article has not been published, how will a user see it to want to leave a comment?

Reply:
While testing the endpoint on postman I unintentionally, entered a slug in the url for an article that was in draft state, yet I was able to comment on the draft article. That's why I added that extra check. But I get your point since readers won't have access to unpublished articles on the frontend then I guess I can take out the check.

Reply:
I think we can leave this